project/plugins.sbt

```diff
@@ -24,6 +24,7 @@ addSbtPlugin("de.heikoseeberger" % "sbt-header" % "5.0.0") // for maintenance of
 addSbtPlugin("com.hpe.sbt" % "sbt-pull-request-validator" % "1.0.0")
 addSbtPlugin("org.scoverage" % "sbt-scoverage" % "1.6.0-M5")
 addSbtPlugin("net.bzzt" % "sbt-reproducible-builds" % "0.21")
+addSbtPlugin("com.dwijnand" % "sbt-dynver" % "3.3.0")
```

Comment:
@dwijnand will you be able to find latest published snapshots in dynver format instead of the timestamp from Play/Lagom jobs?

Reply:
Not sure.  @2m have you seen code for consuming latest released snapshots?

The way we do it now is https://github.com/playframework/playframework/blob/e59ad5f87059d8b607bbdc5ff4a08eae59aeb61d/scripts/scriptLib#L22-L23 I'm not sure if that would work.

Reply:
Hmm, not without changes at least it looks like... let's not merge this before we find a way to do that.

Reply:
I haven't seen it done before. But the oneliner curl scripts in play build look good.

Reply:
that oneliner works for timestamped snapshots, but for dynver versions we'd need something like this to replace the `sort`:

```
#!/usr/bin/perl

$/ = undef;

sub byversion {
  my ($amajor, $aminor, $apatch, $acommits) = ($a =~ /^(\d+)\.(\d+)\.(\d+)\+?(\d*).*/);
  my ($bmajor, $bminor, $bpatch, $bcommits) = ($b =~ /^(\d+)\.(\d+)\.(\d+)\+?(\d*).*/);
  $amajor <=> $bmajor or $aminor <=> $bminor or $apatch <=> $bpatch or $acommits <=> $bcommits
}

print(join("\n", sort byversion (grep { $_ =~ /^\d+\.\d+\.\d+(\+\d+-[0-9a-z]*)?$/ } split("\n", <>))));
```

Reply:
What's causing `sort` to go wrong?  I ask because some of the more recent changes to sbt-dynver were made to help this kind of use case (e.g. https://github.com/dwijnand/sbt-dynver/pull/100 - unreleased).  If there's something else perhaps it's easier to fix in sbt-dynver.

Reply:
`sort` just does a lexicographic sort, so it considers `2.6.2` to be higher than `2.6.11`.

Reply:
`sort -V` does the right thing. We use it in techhub: https://github.com/lightbend/techhub/blob/1264dd41a0457e39072c05e879d987b28172781b/project/BuildUtil.scala#L32

```bash
─╼ echo -e "2.6.11\n2.6.22\n2.6.2" | sort -V                                                                    0
2.6.2
2.6.11
2.6.22
```

Reply:
whoa that's cool, TIL! it does treat milestone and RC versions as 'newer' than their 'bare' version, but that might be acceptable.